src/tox_travis/after.py

```diff
@@ -0,0 +1,150 @@
+"""Add a flag to pause and wait for all Travis jobs to complete."""
+from __future__ import print_function
+import os
+import sys
+import json
+import time
+import py
+
+from tox.config import _split_env as split_env
+try:
+    import urllib.request as urllib2
+except ImportError:
+    import urllib2  # Python 2
+
+from .utils import TRAVIS_FACTORS, parse_dict
+
+
+def travis_after_monkeypatch():
+    """Monkeypatch the Tox session to wait for jobs to finish."""
+    import tox.session
+    real_subcommand_test = tox.session.Session.subcommand_test
+
+    def subcommand_test(self):
+        retcode = real_subcommand_test(self)
+        if retcode == 0 and self.config.option.travis_after:
+            travis_after()  # No need to run if the tests failed anyway
+        return retcode
+    tox.session.Session.subcommand_test = subcommand_test
+
+
+def travis_after():
+    """Wait for all jobs to finish, then exit successfully."""
+    if not os.environ.get('TRAVIS'):
+        print('Not a Travis environment.', file=sys.stderr)
+        sys.exit(31)
```

Comment:
Is it worth defining the exit codes as constants? eg,
```py
UNKNOWN_ENVIRONMENT = 31
...

    sys.exit(UNKNOWN_ENVIRONMENT)
````

Not sure, since a useful error message is printed before the code.